backtrace-library/src/main/java/backtraceio/library/models/metrics/Event.java

```diff
@@ -12,8 +12,11 @@ public abstract class Event {
     @SerializedName("timestamp")
     protected long timestamp;
 
+    /**
+     * The event should always have attributes. Event without attributes won't be processed
+     */
     @SerializedName("attributes")
-    protected Map<String, Object> attributes;
+    protected Map<String, String> attributes = new HashMap<>();
```

Comment:
We can also use synchronized for thread safety : 
`protected final Map<String, String> attributes = Collections.synchronizedMap(new HashMap<>());`

and access it like:

`public Map<String, String> getAttributes() {
        synchronized (attributes) {
            return new HashMap<>(this.attributes);  // Return a copy to prevent modification
        }
    }`

`protected void addAttributes(Map<String, String> attributes) {
        if (attributes == null || attributes.isEmpty()) {
            return;
        }
        synchronized (this.attributes) {
            this.attributes.putAll(attributes);
        }
    }`

Reply:
I think we will never run into a situation where due to not safety thread code we can generate an exception. We're managing metrics and the "send" part.